docs/rules/jsx-sort-props.md

```diff
@@ -75,6 +79,36 @@ When `true`, alphabetical order is not enforced:
 <Hello tel={5555555} name="John" />
 ```
 
+### `reservedFirst`
+
+This can be a boolean or an object option.
+
+When `reservedFirst` is defined, React reserved props (`children`, `dangerouslySetInnerHTML`, `key`, `ref`) must be listed before all other props, but still respecting the alphabetical order:
```

Comment:
Hmmm. Well, all DOM nodes can take the other props, so I think this is ok; I don't actually know what happens if you set `dangerouslySetInnerHTML` on a composite component. Either way, I think users not putting `dangerouslySetInnerHTML` on non-DOM components is something left for a different rule check.